asf_installer_survey/notebooks/section_6_analysis.py

```diff
@@ -0,0 +1,1670 @@
+# **Author:** Elysia Lucas<br>
+# **Date:** 2024-04-04<br>
+# <i>Update:</i> 2024-04-17 (Re-coding of free text responses)
+# <p> 
+# N.B. As of 11.04.2024, only CompanySizeOwner and CompanySizeEmployee outputs with version 2 categorisation are generated.
+
+# %% [markdown]
+# ### Importing and pre-processing
+
+# %%
+with open("data_setup.py") as file:
+    exec(file.read())
+
+with open("free_text_recode.py") as file:
+    exec(file.read())
+
+# %%
+from matplotlib import pyplot as plt
+
+from IPython.core.interactiveshell import InteractiveShell
+InteractiveShell.ast_node_interactivity = "all"
+
+# %% [markdown]
+# ### RQ 6.01 What proportion of survey participants offer repairs, servicing and maintenance to heat pump customers
+# - SQ: 42a, 42b
+# - Sample: Standard_output (employment type categorisation)
+
+# %%
+# Print possible answers
+data[col.q42a].value_counts()
+data[col.q42b].value_counts()
+
+
+# %%
+def condition_q42(x):
+    """
+    Function to be applied to dataframe which creates a new column
+    merging responses from sub-types of the same question.
+
+    Args:
+        x: pandas dataframe containing responses from analytical sample.
+    """
+    
+    # Merge responses
+    if (x[col.q42a] == "Yes") or (x[col.q42b] == "Yes"):
+        return "Yes"
+    
+    elif (x[col.q42a] == "No, but I intend to offer it in the next 12 months") or (x[col.q42b] == "No, but we intend to offer it in the next 12 months"):
+        return "No, but we/I intend to offer it in the next 12 months"
+    
+    elif (x[col.q42a] == "No, and I do not intend to start offering this service in the next 12 months") or (x[col.q42b] == "No, and we do not intend to start offering this service in the next 12 months"):
+        return "No, and we/I do not intend to start offering this service in the next 12 months"
+    
+    elif (x[col.q42a] == "Don't know") or (x[col.q42b] == "Don't know"):
+        return "We/I don't know"
+    
+    # Check if all are captured
+    else:
+        return "Flag" 
+
+# Apply function to each row of dataframe
+data["SQ42"] = data.apply(condition_q42,
+                          axis=1
+                          )
+
+# Specifying desired order for outputs
+ans_42 = ["Yes",
+           "No, but we/I intend to offer it in the next 12 months",
+           "No, and we/I do not intend to start offering this service in the next 12 months",
+           "We/I don't know",
+           "Total"
+           ]
+
+# Generate crosstab dataframe
+df_601_crosstab = crosstable("EmploymentType",
+                             data,
+                             "SQ42",
+                             ans_42
+                             )
+
+# Save to .csv and display
+
+df_601_crosstab.to_csv("../../outputs/section6/csv/601.csv")
```

Comment:
Amazing, thank you for making me aware of this! Will make sure I avoid using relative paths in future.

Reply:
@helloaidank I was wondering what you mean about "after doing this configuration"? I think I may have missed a step when setting up because I am getting this error when trying to import PROJECT_DIR:

![image](https://github.com/nestauk/asf_installer_survey/assets/101257595/642fb052-ed06-4329-ab86-92ca0a4fc8ff)

Reply:
You probably have to do the cookiecutter set-up i.e. : 

Run: `make install`;
Run: `direnv allow`;

And then you will get this functionality with `PROJECT_DIR` and should set up all the requirements for you. Notice that it uses the `setup.py`, so you may need to edit this file to point to `requirements_section_6.txt`.
I am not sure if the set up for installer survey repository had this in mind as I noticed you have to manually run a conda create environment from the `requirements_section_6.txt`, whereas the above creates the conda environment as well.